### PR TITLE
Switching to mktemp instead and quoting where its used.

### PR DESCRIPTION
**Pull request by brimstone**
This should fix the error with tempfile on OS X. @dan9186 please check.

### CODE REVIEW

**Comment by dan9186** [comment]
Unfortunately just mktemp by itself on a Mac doesn't work.  So I've added a check for *nix and reverted it's command to the original.

File: modules/golang/golang [hunk 1]
```diff
@@ -3,8 +3,8 @@ go () {
 	"coverweb")
 		shift
 		repo="${PWD//$GOPATH\/src\//}"
-		t=$(tempfile)
-		command go test $COVERFLAGS -coverprofile=$t "$repo" && go tool cover -html=$t -o coverage.html && unlink $t
+		( islinux && t=$(tempfile) ) || t="$(mktemp -t coverage)"
```
Comment (brimstone) [inline modules/golang/golang]: Can we just use mktemp on both linux and os x?

**Comment by brimstone** [comment]
@dan9186 is this good to merge? @itsamenathan do you have any input on this PR?

**Comment by itsamenathan** [comment]
I think it should check for OSX since that is the edge case we are trying to support.

**Comment by brimstone** [comment]
@itsamenathan I believe line 8 takes into account non linux systems.

**Comment by dan9186** [comment]
The efforts on line 8 would be, if it's *nix do the normal thing otherwise do the edgey thing.

As for good to merge, as far as I know of it is.

**Comment by brimstone** [comment]
@itsamenathan are you ok with this PR as is, or do you want to change it to your liking?

**Comment by itsamenathan** [comment]
I'm good, pull the trigger.

**Pull request merged**